antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    else:
+        raise NotImplementedError(
+            f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+        )
+
+
+def _update_matrices_names(
+    file_study: FileStudy,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> None:
+    """
+    Update the matrix file name according to the new operator.
+
+    Args:
+        file_study: the file study
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Raises:
+        NotImplementedError: if the case is not handled
+    """
+
+    if existing_operator == new_operator:
+        return
+
+    parent_folder_node = file_study.tree.get_node(["input", "bindingconstraints"])
+    assert isinstance(parent_folder_node, FolderNode), f"Node type not handled yet: {type(parent_folder_node)}"
+
+    matrix_paths = _infer_input_and_target_paths(
+        parent_folder_node, binding_constraint_id, existing_operator, new_operator
+    )
+
+    # TODO: due to legacy matrices generation, we need to check if the new matrix file already exists
+    #  and if it does, we need to first remove it before renaming the existing matrix file
+
+    if existing_operator != BindingConstraintOperator.BOTH and new_operator != BindingConstraintOperator.BOTH:
+        (matrix_path,) = matrix_paths.matrix_input_paths
+        (new_matrix_path,) = matrix_paths.matrix_target_paths
+        new_matrix_path.unlink(missing_ok=True)
+        matrix_path.rename(new_matrix_path)
+    elif new_operator == BindingConstraintOperator.BOTH:
+        matrix_path_lt, matrix_path_gt = matrix_paths.matrix_target_paths
+        if existing_operator == BindingConstraintOperator.EQUAL:
+            (matrix_path_eq,) = matrix_paths.matrix_input_paths
+            matrix_path_lt.unlink(missing_ok=True)
+            matrix_path_gt.unlink(missing_ok=True)
+            matrix_path_eq.rename(matrix_path_lt)
+            # copy the matrix lt to gt
+            shutil.copy(matrix_path_lt, matrix_path_gt)
+        elif existing_operator == BindingConstraintOperator.LESS:
+            matrix_path_gt.unlink(missing_ok=True)
+            shutil.copy(matrix_path_lt, matrix_path_gt)
+        elif existing_operator == BindingConstraintOperator.GREATER:
+            matrix_path_lt.unlink(missing_ok=True)
+            shutil.copy(matrix_path_gt, matrix_path_lt)
+    elif existing_operator == BindingConstraintOperator.BOTH:
+        matrix_path_lt, matrix_path_gt = matrix_paths.matrix_input_paths
+        if new_operator == BindingConstraintOperator.EQUAL:
+            # TODO: we may retrieve the mean of the two matrices, but here we just copy the lt matrix
```

Comment:
What does this comment mean ?

Reply:
Actualy when switch from `BOTH` to `EQUAL`, we keep the `LESS` matrix rename to `EQUAL` and remove `GREATER` matrix. Some one might argue shouldn't we compute the mean of both matrices and output it as the `EQUAL` matrix? just to give an example.

Reply:
Okay I see but this is not a TODO, just a comment. If you want to keep track of this you can keep the comment without the TODO